include/sparrow/buffer.hpp

```diff
@@ -686,6 +705,156 @@ namespace sparrow
         erase_at_end(get_data().p_begin);
     }
 
+    template <class T>
+    constexpr auto buffer<T>::insert(const_iterator pos, const T& value) -> iterator
+    {
+        SPARROW_ASSERT_TRUE(cbegin() <= pos);
+        SPARROW_ASSERT_TRUE(pos <= cend());
+        return emplace(pos, value);
+    }
+
+    template <class T>
+    constexpr auto buffer<T>::insert(const_iterator pos, T&& value) -> iterator
+    {
+        SPARROW_ASSERT_TRUE(cbegin() <= pos);
+        SPARROW_ASSERT_TRUE(pos <= cend());
+        return emplace(pos, std::move(value));
+    }
+
+    template <class T>
+    constexpr auto buffer<T>::insert(const_iterator pos, size_type count, const T& value) -> iterator
+    {
+        SPARROW_ASSERT_TRUE(cbegin() <= pos);
+        SPARROW_ASSERT_TRUE(pos <= cend());
+
+        const difference_type __offset = std::distance(cbegin(), pos);
```

Comment:
Double underscore names are reserved by the standard, maybe just change to `offset`?